pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -146,48 +136,51 @@ type ruleCache struct {
 }
 
 func (c *ruleCache) getNetworkPolicies(namespace string) []v1beta1.NetworkPolicy {
-	ret := []v1beta1.NetworkPolicy{}
-	c.policyMapLock.RLock()
-	defer c.policyMapLock.RUnlock()
-	for uid, np := range c.policyMap {
-		if namespace == "" || np.Namespace == namespace {
-			ret = append(ret, *c.buildNetworkPolicyFromRules(uid))
-		}
+	var ret []v1beta1.NetworkPolicy
+	rulesByPolicy := map[v1beta1.NetworkPolicyReference][]*rule{}
+	var objs []interface{}
+	if namespace == "" {
+		objs = c.rules.List()
+	} else {
+		objs, _ = c.rules.ByIndex(namespaceIndex, namespace)
+	}
+	for _, obj := range objs {
+		rule := obj.(*rule)
+		rulesByPolicy[*rule.PolicyRef] = append(rulesByPolicy[*rule.PolicyRef], rule)
+	}
+	for _, rules := range rulesByPolicy {
+		ret = append(ret, *c.buildNetworkPolicyFromRules(rules))
 	}
 	return ret
 }
 
 // getNetworkPolicy looks up and returns the cached NetworkPolicy.
 // nil is returned if the specified NetworkPolicy is not found.
 func (c *ruleCache) getNetworkPolicy(npName, npNamespace string) *v1beta1.NetworkPolicy {
-	var npUID string
-	c.policyMapLock.Lock()
-	defer c.policyMapLock.Unlock()
-	for uid, np := range c.policyMap {
-		if np.Name == npName && np.Namespace == npNamespace {
-			npUID = uid
-			break
+	objs, _ := c.rules.ByIndex(namespaceIndex, npNamespace)
+	var rules []*rule
+	for _, obj := range objs {
+		rule := obj.(*rule)
+		if rule.PolicyRef.Name == npName {
```

Comment:
How is K8s NP and ANP differentiated here? If there were NP and ANP with same ns/name, wouldn't rules for both those policies be added to `rules` still?

Reply:
Good catch, I tried to fix it but it seems not a simple fix as antctl never expects Antrea Policies before. I filed https://github.com/vmware-tanzu/antrea/issues/1262 and listed existing issues. As this interface is consumed by antctl only, I would prefer addressing it with a separate PR for antctl. What do you think?

Reply:
I agree. cc @GraysonWu as he is working on antctl support for ClusterNetworkPolicy. This PR and issue #1262 should be relevant

Reply:
i noticed this too.. but since we have not yet implemented antctl so it won't make sense to handle it in this PR
@tnqn i already raised https://github.com/vmware-tanzu/antrea/issues/927 issue long time back.. will dup your issue to it as Grayson is already assigned the initial PR